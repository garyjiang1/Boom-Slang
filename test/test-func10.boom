def factorial(int n) returns int:
	if n < 0:
		println("invalid input!")
		return -1
	if n == 0:
		return 1
	return n * factorial(n - 1)

println(factorial(5))
println(factorial(0))
println(factorial(-20))
