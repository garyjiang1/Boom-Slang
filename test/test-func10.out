120
1
invalid input!
-1
